app/models/reports/repository.rb

```diff
@@ -155,17 +163,46 @@ def initialize(regions, periods:)
     end
 
     memoize def controlled
+      return controlled_v2 if reporting_schema_v2?
+      science("controlled") do |e|
+        e.use { controlled_v1 }
+        e.try { controlled_v2 }
+        e.raise_on_mismatches = true
+      end
+    end
+
+    def controlled_v1
       region_period_cached_query(__method__) do |entry|
         control_rate_query.controlled(entry.region, entry.period).count
       end
     end
 
+    memoize def controlled_v2
+      regions.each_with_object({}).each do |region, hsh|
+        hsh[region.slug] = control_rate_query_v2.controlled_counts(region, range: active_range(region)).tap { |hsh| hsh.default = 0 }
+      end
```

Comment:
I was intending to avoid this per-region-entry query with v2. We have the ability to query for parent regions in v2, and that is more expressive, and more performant. In other words, 1 query for a block (or district) would be more efficient than 20 (or 200) queries for each facility in it.

Do we plan to explore that approach in a future iteration?

Reply:
Yup, I think that is definitely worth exploring in the future.  My goal here is to get things correct first, and optimize after that.

Reply:
I'm okay with that as well. We should remember that the performance is sub-linear when working with multiple regions in a subquery, and evaluate the performance of the v2 reports after that change has been made.

Also, the correctness aspect might change (or require another verification) since the signature itself might change when we accommodate parent regions. That's part of the reason I am suggesting front-loading that work.